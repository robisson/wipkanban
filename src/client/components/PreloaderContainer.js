import { connect } from "react-redux";
import Preloader from "./Preloader";

const mapStateToProps = ({ userReducer: { user } }) => ({ user });

export default connect(mapStateToProps)(Preloader);
