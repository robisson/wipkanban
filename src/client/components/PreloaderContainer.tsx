import { connect } from "react-redux";
import Preloader from "./Preloader";

const mapStateToProps = ({ userReducer: { showPreloader } }: any) => ({
  showPreloader
});

export default connect(mapStateToProps)(Preloader);
