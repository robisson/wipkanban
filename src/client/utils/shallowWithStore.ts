import { shallow } from "enzyme";

export default (component:any, store:any) => {
  const context = {
    store
  };
  return shallow(component, { context });
};
