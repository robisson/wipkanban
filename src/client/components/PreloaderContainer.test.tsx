import React from "react";
import PreloaderContainer from "./PreloaderContainer";
import shallowWithStore from "../utils/shallowWithStore";
import { createMockStore } from "redux-test-utils";
import { ShallowWrapper } from "enzyme";

describe("<PreloaderContainer /> Component", () => {
  let wrapper: ShallowWrapper;

  it("should to render correctly", () => {
    const testState = {
      userReducer: {
        showPrealoder: true
      }
    };

    const store: Object = createMockStore(testState);
    wrapper = shallowWithStore(<PreloaderContainer />, store);
    expect(wrapper).toEqual({});
    expect(wrapper).toMatchSnapshot();
  });
});
