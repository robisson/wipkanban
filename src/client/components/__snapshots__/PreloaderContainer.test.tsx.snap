// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PreloaderContainer /> Component should to render correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
